### PR TITLE
Refactoring tests: unify assertion of TrackPoints

### PR DESCRIPTION
**Pull request by dennisguse**
The assertions of TrackPoints was implemented several times and done differently.
This makes testing complicated and thus should be improved.

Wait for #848 and then rebase.

### CODE REVIEW

**Comment by dennisguse** [comment]
After merging #848; rebased onto main.
* [ ] Speed is not yet verified (expectation needs to be set)
* [ ] Refactor `TrackPointAssert.expect()` to use TrackPoint constructor directly
* [ ] Check GPX test failure

File: src/main/java/de/dennisguse/opentracks/viewmodels/IntervalStatistics.java [hunk 1]
```diff
@@ -164,9 +164,11 @@ private void add(TrackStatistics trackStatistics, @Nullable TrackPoint lastTrack
             time = time.plus(trackStatistics.getTotalTime());
             gain_m = trackStatistics.hasTotalAltitudeGain() ? trackStatistics.getTotalAltitudeGain() : gain_m;
             loss_m = trackStatistics.hasTotalAltitudeLoss() ? trackStatistics.getTotalAltitudeLoss() : loss_m;
-            if (lastTrackPoint != null) {
-                gain_m = lastTrackPoint.hasAltitudeGain() ? gain_m - lastTrackPoint.getAltitudeGain() : gain_m;
-                loss_m = lastTrackPoint.hasAltitudeLoss() ? loss_m - lastTrackPoint.getAltitudeLoss() : loss_m;
+            if (gain_m != null && lastTrackPoint != null && lastTrackPoint.hasAltitudeGain()) {
```
Comment (rgmf) [inline src/main/java/de/dennisguse/opentracks/viewmodels/IntervalStatistics.java:167]: We have `hasGain` method, why not use it?

```if (hasGain() && lastTrackPoint != null && lastTrackPoint.hasAltitudeGain())```?

**Review by rgmf — COMMENTED** [review]
I think there's a little bug: gain and loss are always unknown "-" (in both track recording layout and intervals fragment).

**Comment by dennisguse** [comment]
@rgmf Thanks - I will check this today!
And I am so happy that the tests are finally working again!

**Comment by dennisguse** [comment]
@rgmf I even did a `=` instead of  `+=`....
Now it's working: after first pressure measurement 0m is shown and gain/loss is updated via temporary TrackStatistics every second.
Now, we can test in an elevator (what I did)...

**Comment by dennisguse** [comment]
@rgmf I should stop working on this PR - it just gets really big.
I just wanted to refactor the tests... :sun_with_face:

**Review by rgmf — CHANGES_REQUESTED** [review]
👍

**Comment by dennisguse** [comment]
@rgmf I am slightly confused - what needs to be changed?

**Comment by rgmf** [comment]
> @rgmf I am slightly confused - what needs to be changed?

If gain/loss unknown issue is fixed then it's ready for me ;)

**Review by rgmf — APPROVED** [review]

**Pull request merged**